packages/grafana-ui/src/components/Box/Box.tsx

```diff
@@ -0,0 +1,54 @@
+import { css } from '@emotion/css';
+import React, { useCallback } from 'react';
+
+import { GrafanaTheme2, ThemeSpacingTokens } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+
+type ExtendSpacing = {
+  vertical?: ThemeSpacingTokens;
+  horizontal?: ThemeSpacingTokens;
+  top?: ThemeSpacingTokens;
+  bottom?: ThemeSpacingTokens;
+  left?: ThemeSpacingTokens;
+  right?: ThemeSpacingTokens;
+};
+type Spacing = ThemeSpacingTokens | ExtendSpacing;
+
+interface BoxProps {
+  margin?: Spacing;
+  padding?: Spacing;
+  backgroundColor?: keyof GrafanaTheme2['colors']['background'];
+}
```

Comment:
On first glance, I'm not a fan of these spacing props. Apart from the (inperceptible) performance implications of constructing new objects on every render (because people will only ever use this like `<Box padding={{top: 1}} />`, I think this API is just awkward, and backs us into a corner for how to resolve responsive values for these.

Instead, I would suggest:
 - padding
 - paddingX, paddingY - i have a subjective preference for x/y over horizontal/vertical, but i don't stand by it 😅 
 - paddingTop, paddingLeft, paddingBottom, paddingRight

I think the existance of the `getSpacing` is kind of a hint that maybe the current API is not quite right.